examples/advanced_pytorch/server.py

```diff
@@ -0,0 +1,110 @@
+from typing import Dict, Optional, Tuple
+from collections import OrderedDict
+import argparse
+from torch.utils.data import DataLoader
+
+import flwr as fl
+import torch
+
+import utils
+
+import warnings
+
+warnings.filterwarnings("ignore")
+
+
+def fit_config(rnd: int):
+    """Return training configuration dict for each round.
+    Keep batch size fixed at 32, perform two rounds of training with one
+    local epoch, increase to two local epochs afterwards.
+    """
+    config = {
+        "batch_size": 16,
+        "local_epochs": 1 if rnd < 2 else 2,
+    }
+    return config
+
+
+def evaluate_config(rnd: int):
+    """Return evaluation configuration dict for each round.
+    Perform five local evaluation steps on each client (i.e., use five
+    batches) during rounds one to three, then increase to ten local
+    evaluation steps.
+    """
+    val_steps = 5 if rnd < 4 else 10
+    return {"val_steps": val_steps}
+
+
+def get_eval_fn(model: torch.nn.Module, toy: bool):
+    """Return an evaluation function for server-side evaluation."""
+
+    # Load data and model here to avoid the overhead of doing it in `evaluate` itself
+    trainset, _, _ = utils.load_data()
+
+    n_train = len(trainset)
+    if toy:
+        # use only 10 samples as validation set
+        valset = torch.utils.data.Subset(trainset, range(n_train - 10, n_train))
+    else:
+        # Use the last 5k training examples as a validation set
+        valset = torch.utils.data.Subset(trainset, range(n_train - 5000, n_train))
+
+    valLoader = DataLoader(valset, batch_size=16)
+    # The `evaluate` function will be called after every round
+    def evaluate(
+        weights: fl.common.Weights,
+    ) -> Optional[Tuple[float, Dict[str, fl.common.Scalar]]]:
+        # Update model with the latest parameters
+        params_dict = zip(model.state_dict().keys(), weights)
+        state_dict = OrderedDict({k: torch.tensor(v) for k, v in params_dict})
+        model.load_state_dict(state_dict, strict=True)
+
+        loss, accuracy = utils.test(model, valLoader)
+        return loss, {"accuracy": accuracy}
+
+    return evaluate
+
+
+def main():
+    """
+    # Load model for
+    # 1. server-side parameter initialization
+    # 2. server-side parameter evaluation
+    """
+
+    # Parse command line argument `partition`
+    parser = argparse.ArgumentParser(description="Flower")
+    parser.add_argument(
+        "--toy",
+        type=bool,
+        default=False,
+        required=False,
+        help="Set to true to use only 10 datasamples for validation. \
+            Useful for testing purposes. Default: False",
+    )
+
```

Comment:
Allow user to decide whether or not to use GPU (cuda). 
```suggestion
    parser.add_argument(
        "--use_cuda",
        type=bool,
        default=False,
        required=False,
        help="Set to true to use GPU. Default: False",
    )
```